[Automation] javac compile fails for com.fasterxml.jackson.module:jackson-module-paranamer:2.22.0

## javac compile on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: javac compile
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `env GVM_TCK_LV=2.22.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean compileTestJava -Pcoordinates=com.fasterxml.jackson.module:jackson-module-paranamer:2.19.0`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/26729804816
Last 300 lines of the log:
```console
========================================================================================
 run-consecutive-tests input
========================================================================================
Coordinates: com.fasterxml.jackson.module:jackson-module-paranamer:2.19.0
Raw versions JSON: [
  "2.22.0"
]
Parsed version count: 1
Parsed versions: 2.22.0
Timeout per Gradle invocation: 5m
========================================================================================
 Testing com.fasterxml.jackson.module:jackson-module-paranamer:2.19.0:2.22.0
========================================================================================
========================================================================================
 com.fasterxml.jackson.module:jackson-module-paranamer:2.19.0:2.22.0 stage native-image run
========================================================================================
Starting stage 'native-image run' for 2.22.0 at 2026-06-01T01:11:07Z
Command: env GVM_TCK_LV=2.22.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean test -Pcoordinates=com.fasterxml.jackson.module:jackson-module-paranamer:2.19.0
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
Starting a Gradle Daemon, 1 busy Daemon could not be reused, use --status for details
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean UP-TO-DATE

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.fasterxml.jackson.module/jackson-module-paranamer/2.19.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 7s
9 actionable tasks: 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest
====================
Testing library: com.fasterxml.jackson.module:jackson-module-paranamer:2.19.0
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.fasterxml.jackson.module/jackson-module-paranamer/2.19.0/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':nativeTestCompile'.
> Could not resolve all dependencies for configuration ':nativeImageTestClasspath'.
   > Could not resolve all dependencies for configuration ':nativeImageTestClasspath'.
      > Could not find com.fasterxml.jackson.core:jackson-core:2.22.0.
        Required by:
            root project 'com.fasterxml.jackson.module.jackson-module-paranamer_tests' > com.fasterxml.jackson.module:jackson-module-paranamer:2.22.0 > com.fasterxml.jackson:jackson-bom:2.22.0
            root project 'com.fasterxml.jackson.module.jackson-module-paranamer_tests' > com.fasterxml.jackson.module:jackson-module-paranamer:2.22.0 > com.fasterxml.jackson.core:jackson-databind:2.22.0

* Try:
> The project declares repositories, effectively ignoring the repositories you have declared in the settings.
   To determine how project repositories are declared, configure your build to fail on project repositories.
   For more information, please refer to https://docs.gradle.org/9.1.0/userguide/declaring_repositories.html#sub:fail_build_on_project_repositories in the Gradle documentation.
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.


Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
8 actionable tasks: 8 up-to-date
BUILD FAILED in 3s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for com.fasterxml.jackson.module:jackson-module-paranamer:2.19.0 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 11s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 2.22.0 at 2026-06-01T01:11:19Z with exit code 1 after 12s
❌ ERROR: 'native-image run' for 2.22.0 failed with exit code 1.
SOFT-FAIL[native-image run][2.22.0][env GVM_TCK_LV=2.22.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean test -Pcoordinates=com.fasterxml.jackson.module:jackson-module-paranamer:2.19.0]
========================================================================================
 com.fasterxml.jackson.module:jackson-module-paranamer:2.19.0:2.22.0 stage javac compile
========================================================================================
Starting stage 'javac compile' for 2.22.0 at 2026-06-01T01:11:19Z
Command: env GVM_TCK_LV=2.22.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean compileTestJava -Pcoordinates=com.fasterxml.jackson.module:jackson-module-paranamer:2.19.0
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.fasterxml.jackson.module/jackson-module-paranamer/2.19.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 884ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava FAILED
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.fasterxml.jackson.module/jackson-module-paranamer/2.19.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':compileTestJava'.
> Could not resolve all dependencies for configuration ':testCompileClasspath'.
   > Could not resolve all dependencies for configuration ':testCompileClasspath'.
      > Could not find com.fasterxml.jackson.core:jackson-core:2.22.0.
        Searched in the following locations:
          - file:/home/runner/.m2/repository/com/fasterxml/jackson/core/jackson-core/2.22.0/jackson-core-2.22.0.pom
          - https://repo.maven.apache.org/maven2/com/fasterxml/jackson/core/jackson-core/2.22.0/jackson-core-2.22.0.pom
        Required by:
            root project 'com.fasterxml.jackson.module.jackson-module-paranamer_tests' > com.fasterxml.jackson.module:jackson-module-paranamer:2.22.0 > com.fasterxml.jackson:jackson-bom:2.22.0
            root project 'com.fasterxml.jackson.module.jackson-module-paranamer_tests' > com.fasterxml.jackson.module:jackson-module-paranamer:2.22.0 > com.fasterxml.jackson.core:jackson-databind:2.22.0

* Try:
> The project declares repositories, effectively ignoring the repositories you have declared in the settings.
   To determine how project repositories are declared, configure your build to fail on project repositories.
   For more information, please refer to https://docs.gradle.org/9.1.0/userguide/declaring_repositories.html#sub:fail_build_on_project_repositories in the Gradle documentation.
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 987ms
8 actionable tasks: 8 up-to-date

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 2s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 2.22.0 at 2026-06-01T01:11:22Z with exit code 1 after 3s
❌ ERROR: 'javac compile' for 2.22.0 failed with exit code 1.
FAILED[javac compile][2.22.0][env GVM_TCK_LV=2.22.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean compileTestJava -Pcoordinates=com.fasterxml.jackson.module:jackson-module-paranamer:2.19.0]
```

## javac compile on ubuntu-latest / JDK latest-ea / mode current-defaults

Failure kind: javac compile
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: current-defaults
Reproducer: `env GVM_TCK_LV=2.22.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean compileTestJava -Pcoordinates=com.fasterxml.jackson.module:jackson-module-paranamer:2.19.0`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/26729804816
Last 300 lines of the log:
```console
========================================================================================
 run-consecutive-tests input
========================================================================================
Coordinates: com.fasterxml.jackson.module:jackson-module-paranamer:2.19.0
Raw versions JSON: [
  "2.22.0"
]
Parsed version count: 1
Parsed versions: 2.22.0
Timeout per Gradle invocation: 5m
========================================================================================
 Testing com.fasterxml.jackson.module:jackson-module-paranamer:2.19.0:2.22.0
========================================================================================
========================================================================================
 com.fasterxml.jackson.module:jackson-module-paranamer:2.19.0:2.22.0 stage native-image run
========================================================================================
Starting stage 'native-image run' for 2.22.0 at 2026-06-01T01:11:04Z
Command: env GVM_TCK_LV=2.22.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean test -Pcoordinates=com.fasterxml.jackson.module:jackson-module-paranamer:2.19.0
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
Starting a Gradle Daemon, 1 busy Daemon could not be reused, use --status for details
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean UP-TO-DATE

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.fasterxml.jackson.module/jackson-module-paranamer/2.19.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 6s
9 actionable tasks: 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest
====================
Testing library: com.fasterxml.jackson.module:jackson-module-paranamer:2.19.0
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.fasterxml.jackson.module/jackson-module-paranamer/2.19.0/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':nativeTestCompile'.
> Could not resolve all dependencies for configuration ':nativeImageTestClasspath'.
   > Could not resolve all dependencies for configuration ':nativeImageTestClasspath'.
      > Could not find com.fasterxml.jackson.core:jackson-core:2.22.0.
        Required by:
            root project 'com.fasterxml.jackson.module.jackson-module-paranamer_tests' > com.fasterxml.jackson.module:jackson-module-paranamer:2.22.0 > com.fasterxml.jackson:jackson-bom:2.22.0
            root project 'com.fasterxml.jackson.module.jackson-module-paranamer_tests' > com.fasterxml.jackson.module:jackson-module-paranamer:2.22.0 > com.fasterxml.jackson.core:jackson-databind:2.22.0

* Try:
> The project declares repositories, effectively ignoring the repositories you have declared in the settings.
   To determine how project repositories are declared, configure your build to fail on project repositories.
   For more information, please refer to https://docs.gradle.org/9.1.0/userguide/declaring_repositories.html#sub:fail_build_on_project_repositories in the Gradle documentation.
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 2s

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
8 actionable tasks: 8 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for com.fasterxml.jackson.module:jackson-module-paranamer:2.19.0 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 11s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 2.22.0 at 2026-06-01T01:11:16Z with exit code 1 after 12s
❌ ERROR: 'native-image run' for 2.22.0 failed with exit code 1.
SOFT-FAIL[native-image run][2.22.0][env GVM_TCK_LV=2.22.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean test -Pcoordinates=com.fasterxml.jackson.module:jackson-module-paranamer:2.19.0]
========================================================================================
 com.fasterxml.jackson.module:jackson-module-paranamer:2.19.0:2.22.0 stage javac compile
========================================================================================
Starting stage 'javac compile' for 2.22.0 at 2026-06-01T01:11:16Z
Command: env GVM_TCK_LV=2.22.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean compileTestJava -Pcoordinates=com.fasterxml.jackson.module:jackson-module-paranamer:2.19.0
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.fasterxml.jackson.module/jackson-module-paranamer/2.19.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 886ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava FAILED
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':compileTestJava'.
> Could not resolve all dependencies for configuration ':testCompileClasspath'.
   > Could not resolve all dependencies for configuration ':testCompileClasspath'.
      > Could not find com.fasterxml.jackson.core:jackson-core:2.22.0.
        Searched in the following locations:
          - file:/home/runner/.m2/repository/com/fasterxml/jackson/core/jackson-core/2.22.0/jackson-core-2.22.0.pom
          - https://repo.maven.apache.org/maven2/com/fasterxml/jackson/core/jackson-core/2.22.0/jackson-core-2.22.0.pom
        Required by:

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.fasterxml.jackson.module/jackson-module-paranamer/2.19.0/build/reports/problems/problems-report.html
            root project 'com.fasterxml.jackson.module.jackson-module-paranamer_tests' > com.fasterxml.jackson.module:jackson-module-paranamer:2.22.0 > com.fasterxml.jackson:jackson-bom:2.22.0
            root project 'com.fasterxml.jackson.module.jackson-module-paranamer_tests' > com.fasterxml.jackson.module:jackson-module-paranamer:2.22.0 > com.fasterxml.jackson.core:jackson-databind:2.22.0

* Try:
> The project declares repositories, effectively ignoring the repositories you have declared in the settings.
   To determine how project repositories are declared, configure your build to fail on project repositories.
   For more information, please refer to https://docs.gradle.org/9.1.0/userguide/declaring_repositories.html#sub:fail_build_on_project_repositories in the Gradle documentation.
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).

> Get more help at https://help.gradle.org.
Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
8 actionable tasks: 8 up-to-date

BUILD FAILED in 930ms

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 2s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 2.22.0 at 2026-06-01T01:11:19Z with exit code 1 after 3s
❌ ERROR: 'javac compile' for 2.22.0 failed with exit code 1.
FAILED[javac compile][2.22.0][env GVM_TCK_LV=2.22.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean compileTestJava -Pcoordinates=com.fasterxml.jackson.module:jackson-module-paranamer:2.19.0]
```

## javac compile on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: javac compile
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `env GVM_TCK_LV=2.22.0 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean compileTestJava -Pcoordinates=com.fasterxml.jackson.module:jackson-module-paranamer:2.19.0`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/26729804816
Last 300 lines of the log:
```console
========================================================================================
 run-consecutive-tests input
========================================================================================
Coordinates: com.fasterxml.jackson.module:jackson-module-paranamer:2.19.0
Raw versions JSON: [
  "2.22.0"
]
Parsed version count: 1
Parsed versions: 2.22.0
Timeout per Gradle invocation: 5m
========================================================================================
 Testing com.fasterxml.jackson.module:jackson-module-paranamer:2.19.0:2.22.0
========================================================================================
========================================================================================
 com.fasterxml.jackson.module:jackson-module-paranamer:2.19.0:2.22.0 stage native-image run
========================================================================================
Starting stage 'native-image run' for 2.22.0 at 2026-06-01T01:11:03Z
Command: env GVM_TCK_LV=2.22.0 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean test -Pcoordinates=com.fasterxml.jackson.module:jackson-module-paranamer:2.19.0
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
Starting a Gradle Daemon, 1 busy Daemon could not be reused, use --status for details
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean UP-TO-DATE

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.fasterxml.jackson.module/jackson-module-paranamer/2.19.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 5s
9 actionable tasks: 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest
====================
Testing library: com.fasterxml.jackson.module:jackson-module-paranamer:2.19.0
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.fasterxml.jackson.module/jackson-module-paranamer/2.19.0/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':nativeTestCompile'.
> Could not resolve all dependencies for configuration ':nativeImageTestClasspath'.
   > Could not resolve all dependencies for configuration ':nativeImageTestClasspath'.
      > Could not find com.fasterxml.jackson.core:jackson-core:2.22.0.
        Required by:
            root project 'com.fasterxml.jackson.module.jackson-module-paranamer_tests' > com.fasterxml.jackson.module:jackson-module-paranamer:2.22.0 > com.fasterxml.jackson:jackson-bom:2.22.0
            root project 'com.fasterxml.jackson.module.jackson-module-paranamer_tests' > com.fasterxml.jackson.module:jackson-module-paranamer:2.22.0 > com.fasterxml.jackson.core:jackson-databind:2.22.0

* Try:
> The project declares repositories, effectively ignoring the repositories you have declared in the settings.

   To determine how project repositories are declared, configure your build to fail on project repositories.
Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.
   For more information, please refer to https://docs.gradle.org/9.1.0/userguide/declaring_repositories.html#sub:fail_build_on_project_repositories in the Gradle documentation.

> Run with --stacktrace option to get the stack trace.
You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.
> Run with --info or --debug option to get more log output.

> Run with --scan to generate a Build Scan (Powered by Develocity).
For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
8 actionable tasks: 8 up-to-date
> Get more help at https://help.gradle.org.

BUILD FAILED in 2s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for com.fasterxml.jackson.module:jackson-module-paranamer:2.19.0 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 9s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 2.22.0 at 2026-06-01T01:11:12Z with exit code 1 after 9s
❌ ERROR: 'native-image run' for 2.22.0 failed with exit code 1.
SOFT-FAIL[native-image run][2.22.0][env GVM_TCK_LV=2.22.0 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean test -Pcoordinates=com.fasterxml.jackson.module:jackson-module-paranamer:2.19.0]
========================================================================================
 com.fasterxml.jackson.module:jackson-module-paranamer:2.19.0:2.22.0 stage javac compile
========================================================================================
Starting stage 'javac compile' for 2.22.0 at 2026-06-01T01:11:12Z
Command: env GVM_TCK_LV=2.22.0 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean compileTestJava -Pcoordinates=com.fasterxml.jackson.module:jackson-module-paranamer:2.19.0
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.fasterxml.jackson.module/jackson-module-paranamer/2.19.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 740ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava FAILED
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.fasterxml.jackson.module/jackson-module-paranamer/2.19.0/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':compileTestJava'.
> Could not resolve all dependencies for configuration ':testCompileClasspath'.
   > Could not resolve all dependencies for configuration ':testCompileClasspath'.
      > Could not find com.fasterxml.jackson.core:jackson-core:2.22.0.
        Searched in the following locations:
          - file:/home/runner/.m2/repository/com/fasterxml/jackson/core/jackson-core/2.22.0/jackson-core-2.22.0.pom
          - https://repo.maven.apache.org/maven2/com/fasterxml/jackson/core/jackson-core/2.22.0/jackson-core-2.22.0.pom
        Required by:
            root project 'com.fasterxml.jackson.module.jackson-module-paranamer_tests' > com.fasterxml.jackson.module:jackson-module-paranamer:2.22.0 > com.fasterxml.jackson:jackson-bom:2.22.0
            root project 'com.fasterxml.jackson.module.jackson-module-paranamer_tests' > com.fasterxml.jackson.module:jackson-module-paranamer:2.22.0 > com.fasterxml.jackson.core:jackson-databind:2.22.0


* Try:
> The project declares repositories, effectively ignoring the repositories you have declared in the settings.
   To determine how project repositories are declared, configure your build to fail on project repositories.
   For more information, please refer to https://docs.gradle.org/9.1.0/userguide/declaring_repositories.html#sub:fail_build_on_project_repositories in the Gradle documentation.
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 748ms
Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
8 actionable tasks: 8 up-to-date

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 2s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 2.22.0 at 2026-06-01T01:11:15Z with exit code 1 after 3s
❌ ERROR: 'javac compile' for 2.22.0 failed with exit code 1.
FAILED[javac compile][2.22.0][env GVM_TCK_LV=2.22.0 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean compileTestJava -Pcoordinates=com.fasterxml.jackson.module:jackson-module-paranamer:2.19.0]
```

--- a/metadata/com.fasterxml.jackson.module/jackson-module-paranamer/2.22.0/reachability-metadata.json
+++ b/metadata/com.fasterxml.jackson.module/jackson-module-paranamer/2.22.0/reachability-metadata.json
@@ -1,0 +1,22 @@
+{
+  "reflection" : [
+    {
+      "condition" : {
+        "typeReached" : "com.fasterxml.jackson.module.paranamer.shaded.AdaptiveParanamer"
+      },
+      "type" : "com.fasterxml.jackson.module.paranamer.shaded.DefaultParanamer",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "com.fasterxml.jackson.module.paranamer.shaded.AdaptiveParanamer"
+      },
+      "type" : "java.lang.StackTraceElement[]"
+    }
+  ]
+}

--- a/metadata/com.fasterxml.jackson.module/jackson-module-paranamer/index.json
+++ b/metadata/com.fasterxml.jackson.module/jackson-module-paranamer/index.json
@@ -1,6 +1,15 @@
 [
   {
     "latest" : true,
+    "metadata-version" : "2.22.0",
+    "tested-versions" : [
+      "2.22.0"
+    ],
+    "allowed-packages" : [
+      "com.fasterxml.jackson.module.paranamer"
+    ]
+  },
+  {
     "metadata-version" : "2.19.0",
     "source-code-url" : "https://repo1.maven.org/maven2/com/fasterxml/jackson/module/jackson-module-paranamer/$version$/jackson-module-paranamer-$version$-sources.jar",
     "repository-url" : "https://github.com/FasterXML/jackson-modules-base",

--- a/metadata/com.fasterxml.jackson.module/jackson-module-paranamer/index.json
+++ b/metadata/com.fasterxml.jackson.module/jackson-module-paranamer/index.json
@@ -2,6 +2,11 @@
   {
     "latest" : true,
     "metadata-version" : "2.22.0",
+    "source-code-url" : "https://repo1.maven.org/maven2/com/fasterxml/jackson/module/jackson-module-paranamer/$version$/jackson-module-paranamer-$version$-sources.jar",
+    "repository-url" : "https://github.com/FasterXML/jackson-modules-base",
+    "test-code-url" : "https://github.com/FasterXML/jackson-modules-base/tree/jackson-modules-base-$version$/paranamer/src/test/java",
+    "documentation-url" : "https://repo1.maven.org/maven2/com/fasterxml/jackson/module/jackson-module-paranamer/$version$/jackson-module-paranamer-$version$-javadoc.jar",
+    "description" : "Jackson Module Paranamer adds annotation introspection support that lets Jackson discover constructor and factory method parameter names through the Paranamer library. It helps Jackson databind deserialize types whose creators rely on parameter names when explicit Jackson annotations are not present.",
     "tested-versions" : [
       "2.22.0"
     ],

--- a/stats/com.fasterxml.jackson.module/jackson-module-paranamer/2.22.0/execution-metrics.json
+++ b/stats/com.fasterxml.jackson.module/jackson-module-paranamer/2.22.0/execution-metrics.json
@@ -1,0 +1,111 @@
+{
+  "fix_javac_fail:2026-06-05": {
+    "timestamp": "2026-06-05T20:00:23.823365Z",
+    "library": "com.fasterxml.jackson.module:jackson-module-paranamer:2.22.0",
+    "starting_commit": "c8bb8a7d5239852870f2522e517086883209bb96",
+    "ending_commit": "492063e9bfdf38c804e07d13e8e52fb81f3cae0a",
+    "previous_library": "com.fasterxml.jackson.module:jackson-module-paranamer:2.19.0",
+    "strategy_name": "javac_iterative_with_coverage_sources_pi_gpt-5.5",
+    "agent": "pi",
+    "model": "oca/gpt-5.5",
+    "status": "success",
+    "stats": {
+      "version": "2.22.0",
+      "dynamicAccess": {
+        "breakdown": {
+          "reflection": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 4,
+            "totalCalls": 4
+          },
+          "resources": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 2,
+            "totalCalls": 2
+          }
+        },
+        "coverageRatio": 1.0,
+        "coveredCalls": 6,
+        "totalCalls": 6
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 1421,
+          "missed": 1453,
+          "ratio": 0.494433,
+          "total": 2874
+        },
+        "line": {
+          "covered": 333,
+          "missed": 357,
+          "ratio": 0.482609,
+          "total": 690
+        },
+        "method": {
+          "covered": 45,
+          "missed": 71,
+          "ratio": 0.387931,
+          "total": 116
+        }
+      }
+    },
+    "previous_library_stats": {
+      "version": "2.19.0",
+      "dynamicAccess": {
+        "breakdown": {
+          "reflection": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 4,
+            "totalCalls": 4
+          },
+          "resources": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 2,
+            "totalCalls": 2
+          }
+        },
+        "coverageRatio": 1.0,
+        "coveredCalls": 6,
+        "totalCalls": 6
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 1415,
+          "missed": 1459,
+          "ratio": 0.492345,
+          "total": 2874
+        },
+        "line": {
+          "covered": 331,
+          "missed": 359,
+          "ratio": 0.47971,
+          "total": 690
+        },
+        "method": {
+          "covered": 45,
+          "missed": 71,
+          "ratio": 0.387931,
+          "total": 116
+        }
+      }
+    },
+    "metrics": {
+      "input_tokens_used": 32227,
+      "cached_input_tokens_used": 190464,
+      "output_tokens_used": 2278,
+      "iterations": 1,
+      "cost_usd": 0.1635,
+      "tested_library_loc": 333,
+      "metadata_entries": 3,
+      "test_only_metadata_entries": 15,
+      "previous_library_metadata_entries": 3,
+      "previous_library_test_only_metadata_entries": 13,
+      "code_coverage_percent": 48.26,
+      "previous_library_coverage_percent": 47.97
+    },
+    "artifacts": {
+      "test_file": "tests/src/com.fasterxml.jackson.module/jackson-module-paranamer/2.22.0/src/test/java/com_fasterxml_jackson_module/jackson_module_paranamer/BytecodeReadingParanamerTest.java",
+      "metadata_file": "metadata/com.fasterxml.jackson.module/jackson-module-paranamer/2.22.0/reachability-metadata.json"
+    }
+  }
+}

--- a/stats/com.fasterxml.jackson.module/jackson-module-paranamer/2.22.0/stats.json
+++ b/stats/com.fasterxml.jackson.module/jackson-module-paranamer/2.22.0/stats.json
@@ -1,0 +1,42 @@
+{
+  "versions" : [ {
+    "version" : "2.22.0",
+    "dynamicAccess" : {
+      "breakdown" : {
+        "reflection" : {
+          "coverageRatio" : 1.0,
+          "coveredCalls" : 4,
+          "totalCalls" : 4
+        },
+        "resources" : {
+          "coverageRatio" : 1.0,
+          "coveredCalls" : 2,
+          "totalCalls" : 2
+        }
+      },
+      "coverageRatio" : 1.0,
+      "coveredCalls" : 6,
+      "totalCalls" : 6
+    },
+    "libraryCoverage" : {
+      "instruction" : {
+        "covered" : 1421,
+        "missed" : 1453,
+        "ratio" : 0.494433,
+        "total" : 2874
+      },
+      "line" : {
+        "covered" : 333,
+        "missed" : 357,
+        "ratio" : 0.482609,
+        "total" : 690
+      },
+      "method" : {
+        "covered" : 45,
+        "missed" : 71,
+        "ratio" : 0.387931,
+        "total" : 116
+      }
+    }
+  } ]
+}

--- a/tests/src/com.fasterxml.jackson.module/jackson-module-paranamer/2.22.0/.gitignore
+++ b/tests/src/com.fasterxml.jackson.module/jackson-module-paranamer/2.22.0/.gitignore
@@ -1,0 +1,4 @@
+gradlew.bat
+gradlew
+gradle/
+build/

--- a/tests/src/com.fasterxml.jackson.module/jackson-module-paranamer/2.22.0/build.gradle
+++ b/tests/src/com.fasterxml.jackson.module/jackson-module-paranamer/2.22.0/build.gradle
@@ -1,0 +1,27 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+plugins {
+    id "org.graalvm.internal.tck"
+}
+
+String libraryVersion = tck.testedLibraryVersion.get()
+
+dependencies {
+    testImplementation "com.fasterxml.jackson.module:jackson-module-paranamer:$libraryVersion"
+    testImplementation 'org.assertj:assertj-core:3.22.0'
+}
+
+graalvmNative {
+    agent {
+        defaultMode = "conditional"
+        modes {
+            conditional {
+                userCodeFilterPath = "user-code-filter.json"
+            }
+        }
+    }
+}

--- a/tests/src/com.fasterxml.jackson.module/jackson-module-paranamer/2.22.0/gradle.properties
+++ b/tests/src/com.fasterxml.jackson.module/jackson-module-paranamer/2.22.0/gradle.properties
@@ -1,0 +1,6 @@
+# Optional explicit tested library coordinates (format: group:artifact:version).
+# If omitted, fallback resolution in org.graalvm.internal.tck.gradle still uses
+# environment variable GVM_TCK_LC and then this property.
+library.coordinates = com.fasterxml.jackson.module:jackson-module-paranamer:2.22.0
+library.version = 2.22.0
+metadata.dir = com.fasterxml.jackson.module/jackson-module-paranamer/2.22.0/

--- a/tests/src/com.fasterxml.jackson.module/jackson-module-paranamer/2.22.0/settings.gradle
+++ b/tests/src/com.fasterxml.jackson.module/jackson-module-paranamer/2.22.0/settings.gradle
@@ -1,0 +1,20 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+pluginManagement {
+    def tckPath = Objects.requireNonNullElse(
+            System.getenv("GVM_TCK_TCKDIR"),
+            "../../../../tck-build-logic"
+    )
+    includeBuild(tckPath)
+    repositories { mavenLocal(); gradlePluginPortal() }
+}
+
+plugins {
+    id "org.graalvm.internal.tck-settings" version "1.0.0-SNAPSHOT"
+}
+
+rootProject.name = 'com.fasterxml.jackson.module.jackson-module-paranamer_tests'

--- a/tests/src/com.fasterxml.jackson.module/jackson-module-paranamer/2.22.0/src/test/java/com_fasterxml_jackson_module/jackson_module_paranamer/AdaptiveParanamerTest.java
+++ b/tests/src/com.fasterxml.jackson.module/jackson-module-paranamer/2.22.0/src/test/java/com_fasterxml_jackson_module/jackson_module_paranamer/AdaptiveParanamerTest.java
@@ -1,0 +1,29 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package com_fasterxml_jackson_module.jackson_module_paranamer;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.fasterxml.jackson.module.paranamer.shaded.AdaptiveParanamer;
+import java.lang.reflect.Constructor;
+import org.junit.jupiter.api.Test;
+
+public class AdaptiveParanamerTest {
+    @Test
+    void defaultConstructorIncludesReflectionBasedParameterNameLookup() throws Exception {
+        AdaptiveParanamer paranamer = new AdaptiveParanamer();
+        Constructor<ConstructorParameterFixture> constructor = ConstructorParameterFixture.class.getDeclaredConstructor(
+                String.class, int.class);
+
+        String[] parameterNames = paranamer.lookupParameterNames(constructor, false);
+
+        assertThat(parameterNames).containsExactly("text", "repeatCount");
+    }
+
+    public record ConstructorParameterFixture(String text, int repeatCount) {
+    }
+}

--- a/tests/src/com.fasterxml.jackson.module/jackson-module-paranamer/2.22.0/src/test/java/com_fasterxml_jackson_module/jackson_module_paranamer/AdaptiveParanamerTest.java
+++ b/tests/src/com.fasterxml.jackson.module/jackson-module-paranamer/2.22.0/src/test/java/com_fasterxml_jackson_module/jackson_module_paranamer/AdaptiveParanamerTest.java
@@ -9,21 +9,27 @@ package com_fasterxml_jackson_module.jackson_module_paranamer;
 import static org.assertj.core.api.Assertions.assertThat;
 
 import com.fasterxml.jackson.module.paranamer.shaded.AdaptiveParanamer;
-import java.lang.reflect.Constructor;
+import java.lang.reflect.Method;
 import org.junit.jupiter.api.Test;
 
 public class AdaptiveParanamerTest {
     @Test
-    void defaultConstructorIncludesReflectionBasedParameterNameLookup() throws Exception {
+    void defaultConstructorReadsParameterNamesFromAvailableStrategies() throws Exception {
         AdaptiveParanamer paranamer = new AdaptiveParanamer();
-        Constructor<ConstructorParameterFixture> constructor = ConstructorParameterFixture.class.getDeclaredConstructor(
-                String.class, int.class);
+        Method method = MethodParameterFixture.class.getDeclaredMethod("format", String.class, int.class);
 
-        String[] parameterNames = paranamer.lookupParameterNames(constructor, false);
+        String[] parameterNames = paranamer.lookupParameterNames(method, false);
 
-        assertThat(parameterNames).containsExactly("text", "repeatCount");
+        assertThat(parameterNames).containsExactly("message", "repeatCount");
     }
 
-    public record ConstructorParameterFixture(String text, int repeatCount) {
+    private static final class MethodParameterFixture {
+        String format(String message, int repeatCount) {
+            StringBuilder result = new StringBuilder();
+            for (int index = 0; index < repeatCount; index++) {
+                result.append(message);
+            }
+            return result.toString();
+        }
     }
 }

--- a/tests/src/com.fasterxml.jackson.module/jackson-module-paranamer/2.22.0/src/test/java/com_fasterxml_jackson_module/jackson_module_paranamer/BytecodeReadingParanamerTest.java
+++ b/tests/src/com.fasterxml.jackson.module/jackson-module-paranamer/2.22.0/src/test/java/com_fasterxml_jackson_module/jackson_module_paranamer/BytecodeReadingParanamerTest.java
@@ -1,0 +1,63 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package com_fasterxml_jackson_module.jackson_module_paranamer;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.fasterxml.jackson.module.paranamer.shaded.BytecodeReadingParanamer;
+import java.io.InputStream;
+import java.lang.reflect.Method;
+import org.junit.jupiter.api.Test;
+
+public class BytecodeReadingParanamerTest {
+    @Test
+    void readsParameterNamesFromAvailableClassResource() throws Exception {
+        BytecodeReadingParanamer paranamer = new BytecodeReadingParanamer();
+        Method method = ParameterNameFixture.class.getDeclaredMethod("format", String.class, int.class);
+        assertThatClassResourceIsAvailable(method.getDeclaringClass());
+
+        String[] parameterNames = paranamer.lookupParameterNames(method, false);
+
+        assertThat(parameterNames).containsExactly("message", "repeatCount");
+    }
+
+    @Test
+    void returnsEmptyNamesWhenGeneratedClassHasNoBytecodeResource() throws Exception {
+        BytecodeReadingParanamer paranamer = new BytecodeReadingParanamer();
+        ParameterizedOperation operation = value -> value.length();
+        Method method = operation.getClass().getDeclaredMethod("apply", String.class);
+        assertThat(method.getDeclaringClass()).isSameAs(operation.getClass());
+
+        String[] parameterNames = paranamer.lookupParameterNames(method, false);
+
+        assertThat(parameterNames).isEmpty();
+    }
+
+    private interface ParameterizedOperation {
+        int apply(String value);
+    }
+
+    private static void assertThatClassResourceIsAvailable(Class<?> type) throws Exception {
+        try (InputStream resource = type.getClassLoader().getResourceAsStream(classResourceName(type))) {
+            assertThat(resource).isNotNull();
+        }
+    }
+
+    private static String classResourceName(Class<?> type) {
+        return type.getName().replace('.', '/') + ".class";
+    }
+
+    private static final class ParameterNameFixture {
+        String format(String message, int repeatCount) {
+            StringBuilder result = new StringBuilder();
+            for (int index = 0; index < repeatCount; index++) {
+                result.append(message);
+            }
+            return result.toString();
+        }
+    }
+}

--- a/tests/src/com.fasterxml.jackson.module/jackson-module-paranamer/2.22.0/src/test/java/com_fasterxml_jackson_module/jackson_module_paranamer/DefaultParanamerTest.java
+++ b/tests/src/com.fasterxml.jackson.module/jackson-module-paranamer/2.22.0/src/test/java/com_fasterxml_jackson_module/jackson_module_paranamer/DefaultParanamerTest.java
@@ -1,0 +1,29 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package com_fasterxml_jackson_module.jackson_module_paranamer;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.fasterxml.jackson.module.paranamer.shaded.DefaultParanamer;
+import java.lang.reflect.Constructor;
+import org.junit.jupiter.api.Test;
+
+public class DefaultParanamerTest {
+    @Test
+    void readsParameterNamesFromReflectionMetadata() throws Exception {
+        DefaultParanamer paranamer = new DefaultParanamer();
+        Constructor<ConstructorParameterFixture> constructor = ConstructorParameterFixture.class.getDeclaredConstructor(
+                String.class, int.class);
+
+        String[] parameterNames = paranamer.lookupParameterNames(constructor);
+
+        assertThat(parameterNames).containsExactly("text", "repeatCount");
+    }
+
+    public record ConstructorParameterFixture(String text, int repeatCount) {
+    }
+}

--- a/tests/src/com.fasterxml.jackson.module/jackson-module-paranamer/2.22.0/src/test/java/com_fasterxml_jackson_module/jackson_module_paranamer/LegacyParanamerTest.java
+++ b/tests/src/com.fasterxml.jackson.module/jackson-module-paranamer/2.22.0/src/test/java/com_fasterxml_jackson_module/jackson_module_paranamer/LegacyParanamerTest.java
@@ -1,0 +1,40 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package com_fasterxml_jackson_module.jackson_module_paranamer;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.fasterxml.jackson.module.paranamer.shaded.LegacyParanamer;
+import java.lang.reflect.Method;
+import org.junit.jupiter.api.Test;
+
+public class LegacyParanamerTest {
+    @Test
+    void readsParameterNamesFromEmbeddedParanamerDataField() throws Exception {
+        LegacyParanamer paranamer = new LegacyParanamer();
+        Method method = MethodParameterFixture.class.getDeclaredMethod("combine", String.class, int.class);
+
+        String[] parameterNames = paranamer.lookupParameterNames(method);
+
+        assertThat(parameterNames).containsExactly("text", "repeatCount");
+    }
+
+    public static final class MethodParameterFixture {
+        public static final String __PARANAMER_DATA = """
+                v1.0
+                combine java.lang.String,int text,repeatCount
+                """;
+
+        public String combine(String text, int repeatCount) {
+            StringBuilder result = new StringBuilder();
+            for (int index = 0; index < repeatCount; index++) {
+                result.append(text);
+            }
+            return result.toString();
+        }
+    }
+}

--- a/tests/src/com.fasterxml.jackson.module/jackson-module-paranamer/2.22.0/src/test/resources/META-INF/native-image/reachability-metadata.json
+++ b/tests/src/com.fasterxml.jackson.module/jackson-module-paranamer/2.22.0/src/test/resources/META-INF/native-image/reachability-metadata.json
@@ -1,0 +1,96 @@
+{
+  "reflection" : [
+    {
+      "condition" : {
+        "typeReached" : "com.fasterxml.jackson.module.paranamer.shaded.DefaultParanamer"
+      },
+      "type" : "com_fasterxml_jackson_module.jackson_module_paranamer.DefaultParanamerTest$ConstructorParameterFixture",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [
+            "java.lang.String",
+            "int"
+          ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "com_fasterxml_jackson_module.jackson_module_paranamer.BytecodeReadingParanamerTest"
+      },
+      "type" : "com_fasterxml_jackson_module.jackson_module_paranamer.BytecodeReadingParanamerTest$ParameterNameFixture"
+    },
+    {
+      "condition" : {
+        "typeReached" : "com_fasterxml_jackson_module.jackson_module_paranamer.DefaultParanamerTest"
+      },
+      "type" : "com_fasterxml_jackson_module.jackson_module_paranamer.DefaultParanamerTest$ConstructorParameterFixture"
+    },
+    {
+      "condition" : {
+        "typeReached" : "com.fasterxml.jackson.module.paranamer.shaded.LegacyParanamer"
+      },
+      "type" : "com_fasterxml_jackson_module.jackson_module_paranamer.LegacyParanamerTest$MethodParameterFixture",
+      "fields" : [
+        {
+          "name" : "__PARANAMER_DATA"
+        }
+      ],
+      "methods" : [
+        {
+          "name" : "combine",
+          "parameterTypes" : [
+            "java.lang.String",
+            "int"
+          ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "com_fasterxml_jackson_module.jackson_module_paranamer.BytecodeReadingParanamerTest"
+      },
+      "type" : {
+        "lambda" : {
+          "declaringClass" : "com_fasterxml_jackson_module.jackson_module_paranamer.BytecodeReadingParanamerTest",
+          "interfaces" : [
+            "com_fasterxml_jackson_module.jackson_module_paranamer.BytecodeReadingParanamerTest$ParameterizedOperation"
+          ]
+        }
+      }
+    }
+  ],
+  "resources" : [
+    {
+      "condition" : {
+        "typeReached" : "com_fasterxml_jackson_module.jackson_module_paranamer.BytecodeReadingParanamerTest"
+      },
+      "glob" : "META-INF/services/org.assertj.core.configuration.Configuration"
+    },
+    {
+      "condition" : {
+        "typeReached" : "com_fasterxml_jackson_module.jackson_module_paranamer.BytecodeReadingParanamerTest"
+      },
+      "glob" : "META-INF/services/org.assertj.core.presentation.Representation"
+    },
+    {
+      "condition" : {
+        "typeReached" : "com_fasterxml_jackson_module.jackson_module_paranamer.BytecodeReadingParanamerTest"
+      },
+      "glob" : "com/fasterxml/jackson/module/paranamer/shaded/com_fasterxml_jackson_module/jackson_module_paranamer/BytecodeReadingParanamerTest$$Lambda/0x00000000141ae470.class"
+    },
+    {
+      "condition" : {
+        "typeReached" : "com_fasterxml_jackson_module.jackson_module_paranamer.BytecodeReadingParanamerTest"
+      },
+      "glob" : "com_fasterxml_jackson_module/jackson_module_paranamer/BytecodeReadingParanamerTest$$Lambda/0x00000000141ae470.class"
+    },
+    {
+      "condition" : {
+        "typeReached" : "com_fasterxml_jackson_module.jackson_module_paranamer.BytecodeReadingParanamerTest"
+      },
+      "glob" : "com_fasterxml_jackson_module/jackson_module_paranamer/BytecodeReadingParanamerTest$ParameterNameFixture.class"
+    }
+  ]
+}

--- a/tests/src/com.fasterxml.jackson.module/jackson-module-paranamer/2.22.0/src/test/resources/META-INF/native-image/reachability-metadata.json
+++ b/tests/src/com.fasterxml.jackson.module/jackson-module-paranamer/2.22.0/src/test/resources/META-INF/native-image/reachability-metadata.json
@@ -2,6 +2,12 @@
   "reflection" : [
     {
       "condition" : {
+        "typeReached" : "com_fasterxml_jackson_module.jackson_module_paranamer.AdaptiveParanamerTest"
+      },
+      "type" : "com_fasterxml_jackson_module.jackson_module_paranamer.AdaptiveParanamerTest$MethodParameterFixture"
+    },
+    {
+      "condition" : {
         "typeReached" : "com.fasterxml.jackson.module.paranamer.shaded.DefaultParanamer"
       },
       "type" : "com_fasterxml_jackson_module.jackson_module_paranamer.DefaultParanamerTest$ConstructorParameterFixture",
@@ -62,6 +68,12 @@
     }
   ],
   "resources" : [
+    {
+      "condition" : {
+        "typeReached" : "com_fasterxml_jackson_module.jackson_module_paranamer.AdaptiveParanamerTest"
+      },
+      "glob" : "com_fasterxml_jackson_module/jackson_module_paranamer/AdaptiveParanamerTest$MethodParameterFixture.class"
+    },
     {
       "condition" : {
         "typeReached" : "com_fasterxml_jackson_module.jackson_module_paranamer.BytecodeReadingParanamerTest"

--- a/tests/src/com.fasterxml.jackson.module/jackson-module-paranamer/2.22.0/user-code-filter.json
+++ b/tests/src/com.fasterxml.jackson.module/jackson-module-paranamer/2.22.0/user-code-filter.json
@@ -1,0 +1,13 @@
+{
+  "rules" : [
+    {
+      "excludeClasses" : "**"
+    },
+    {
+      "includeClasses" : "com.fasterxml.jackson.module.paranamer.**"
+    },
+    {
+      "includeClasses" : "com_fasterxml_jackson_module.jackson_module_paranamer.**"
+    }
+  ]
+}

